### PR TITLE
[doc] add lm doc

### PR DESCRIPTION
**Pull request by robin1001**

### CODE REVIEW

File: docs/lm.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+There are two main parts of CTC WFST based search.
+
+The first is building the decoding graph, which is to compose the model unit T, lexicon L and language model G into one unified graph TLG. And in which:
+1. T is the model unit in E2E training. Typically it's char in Chinese, char or BPE in english.
+2. L is lexicon, the lexicon is very simple. What we need to do is just split a word into its modeling unit sequence.
+For example, the word "我们" is split into two chars "我 们", and the word "APPLE" is split into five letters "A P P L E" .
+We can see there is no phonemes and there is no need design pronunciation on purpose.
+3. G is the language model, namely compling the n-gram to standard WFST representation.
+
+The second is decoder, which is same to the traditional decoder, which use standard viterbi beam search algorithm in decoding.
```
Comment (pengzhendong) [inline docs/lm.md:50]: is the decoder
the same as
uses the standard Viterbi

File: docs/lm.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Implementation
+
+WeNet draws on the decoder and related tools in Kaldi to support LM and WFST based decocding.
```
Comment (pengzhendong) [inline docs/lm.md:54]: decoding

File: docs/lm.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+For ease of use and keep independence, we directly migrated the code related to decoding in Kaldi to [this directory](https://github.com/mobvoi/wenet/tree/main/runtime/core/kaldi) in WeNet runtime.
+And modify and organize according to the following principles:
+1. In order to minimize changes, the migrated code remains the same directory structure as the original.
```
Comment (pengzhendong) [inline docs/lm.md:57]: To minimize

File: docs/lm.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+2. We use GLOG to replace the log system in Kaldi.
+3. We modify the code format to meet the lint requirements of the code style in WeNet.
+
+The core code is https://github.com/mobvoi/wenet/blob/main/runtime/core/decoder/ctc_wfst_beam_search.cc,
+which wraps the LatticeFasterDecoder in Kaldi.
+And we use blank frame skipping to speed up decoding.
+
+In addition, WeNet also migrated related tools for building the decoding graph,
+such as arpa2fst, fstdeterminizestar, fsttablecompose, fstminimizeencoded and other tools.
```
Comment (pengzhendong) [inline docs/lm.md:66]: fstminimizeencoded, and other tools

File: docs/lm.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+So all the tools related to LM are built-in tools, and can be used out of box.
```
Comment (pengzhendong) [inline docs/lm.md:67]: built-in tools and can be used out of the box.

File: docs/lm.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+
+## Results
+
+We get consistent gain(3%~10%) on different datasets,
+including aishell, aishell2 and librispeech,
```
Comment (pengzhendong) [inline docs/lm.md:73]: aishell2, and librispeech

File: docs/lm.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+please go to the corresponding example dataset for the details.
+
+## How to use?
+
+Here is an example from aishell, which shows how to prepare dict, how to train LM,
```
Comment (pengzhendong) [inline docs/lm.md:78]: prepare the dictionary, how to train the LM

**Review by placebokkk — APPROVED** [review]

**Pull request merged**